### PR TITLE
Feature/fix enforce limits

### LINKED ISSUE #67: Fix position_control_sample.launch
## Abstract
Specify PID gain in position_control_sample.launch.

### LINKED ISSUE #68: [vesc_hw_interface]enforceLimits() using fixed control period(100Hz)
## Abstract
I'm not sure if there is any change in functionality, but looking at the source code, it is odd, so I'm making an issue.

https://github.com/sbgisen/vesc/blob/d96dcc2d81419f33647b37b249769c0101b7b7f5/vesc_hw_interface/src/vesc_hw_interface.cpp#L244
https://github.com/sbgisen/vesc/blob/d96dcc2d81419f33647b37b249769c0101b7b7f5/vesc_hw_interface/src/vesc_hw_interface.cpp#L299-L302

Control rate is specified as a parameter in vesc_servo_controller and vesc_wheel_controller
https://github.com/sbgisen/vesc/blob/d96dcc2d81419f33647b37b249769c0101b7b7f5/vesc_hw_interface/src/vesc_servo_controller.cpp#L55-L56

https://github.com/sbgisen/vesc/blob/d96dcc2d81419f33647b37b249769c0101b7b7f5/vesc_hw_interface/src/vesc_wheel_controller.cpp#L39

### PR DESCRIPTION
**Pull request by nyxrobotics**
## Change Summary
Fixed a bug in enforceLimits.
Changed the default value of control cycle.

- Fix #67 
- Fix #68 
- Reference: #69

### CODE REVIEW

File: vesc_hw_interface/launch/test.xacro [hunk 1]
```diff
@@ -7,7 +7,8 @@
     <child link="vesc_link"/>
     <origin xyz="0 0 0" rpy="0 0 0"/>
     <axis xyz="0 0 1"/>
-    <limit effort="100" lower="-100" upper="100" velocity="100"/>
+    <limit effort="100" lower="-100" upper="100" velocity="1000"/>
+    <safety_controller k_velocity="10" k_position="15" soft_lower_limit="-2.0" soft_upper_limit="0.5" />
```
Comment (jsupratman13) [inline vesc_hw_interface/launch/test.xacro:11]: Is this a test parameter that you would like to test in the future, or is this parameter tested already?
If the parameter is tested already, do you have any recommended PID gains to test it?

**Review by jsupratman13 — CHANGES_REQUESTED** [review]
At this point, I think we should just deprecate `write()` and `read()` function and use `write(const ros::Time&, const::rosDuration&)` and `read(const ros::Time&, const::rosDuration&)` function instead

File: vesc_hw_interface/launch/test.xacro (hunk 1 above)
Comment (nyxrobotics) [reply vesc_hw_interface/launch/test.xacro:11]: I have tested that it does not rotate faster than the speed set in ```k_velocity``` of safety_controller.(with wheel controller)
I don't know what the other parameters are.
I tested movero hub motor with the params below.

```yaml
    <rosparam>
      joint_name: vesc_joint
      command_mode: velocity_duty
      joint_type: continuous
      port: /dev/ttyACM0
      num_rotor_poles: 20
      gear_ratio: 0.230769
      torque_const: 1
      motor:
        Kp: 0.02
        Ki: 0.01
        Kd: 0.03
        i_clamp: 0.2
        duty_limiter: 0.3
        antiwindup: true
        control_rate: 100
    </rosparam>
```

Comment (jsupratman13) [reply vesc_hw_interface/launch/test.xacro:11]: Sorry for the lack of clarification; I meant the PID gain given through ros_controller.

I believe safety_controller tags are used if we have something set up like vesc hardware interface to JointVelocityInterface but control with PositionController.
ex:
```xml
  <arg name="model" default="$(find vesc_hw_interface)/launch/test.xacro"/>
  <param name="robot_description" command="$(find xacro)/xacro $(arg model)"/>
  <!-- Boot hardware interfaces -->
  <node name="vesc_hw_interface_node" pkg="vesc_hw_interface" type="vesc_hw_interface_node" output="screen">
    <rosparam>
      joint_name: vesc_joint
      command_mode: velocity
      port: /dev/ttyUSB0
      num_rotor_poles: 20
      gear_ratio: 1
      torque_const: 1
    </rosparam>
  </node>

  <!-- Boot ros_controllers -->
  <rosparam>
    joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 100

    joint_position_controller:
      type: position_controllers/JointVelocityController
      publish_rate: 100
      base_frame_id: base_link
      joint: vesc_joint
      pid: {p: 100.0, i: 0.01, d: 10.0}
  </rosparam>
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="joint_state_controller joint_position_controller"/>

```

Comment (jsupratman13) [reply vesc_hw_interface/launch/test.xacro:11]: Note that VescServoController and VescWheelController is setup as JointPositionController, JointVelocityController respectively thus, safety_controller tag is not used.

**Review by jsupratman13 — APPROVED** [review]

File: vesc_hw_interface/launch/test.xacro (hunk 1 above)
Comment (jsupratman13) [reply vesc_hw_interface/launch/test.xacro:11]: I'm leaving the changes here since this is just a sample parameter. Will reopen/tune if necessary.

**Pull request merged**